session_2/student_registry/db/migrate/20210114041229_create_students.rb

```diff
@@ -0,0 +1,13 @@
+class CreateStudents < ActiveRecord::Migration[6.1]
+  def change
+    create_table :students do |t|
+      t.string :name
+      t.string :roll_number
+      t.string :branch
+      t.decimal :cgpa, :precision => 3, :scale => 2
```

Comment:
While precision of 3 is better, can we save students with cgpa `10.00`?

It can work as follows:
- If cgpa is less than 10, approximate to first two decimal digits.
- if cgpa is 10, store `10.0` and not `10.00`

Can you override `cgpa=` function to make it work?

The function should take any floating value, throw `ArgumentError` exception if cgpa > 10 or < 0 and store it appropriately.

https://api.rubyonrails.org/classes/ActiveRecord/Base.html#class-ActiveRecord::Base-label-Overwriting+default+accessors

Do try your hand at it! :v: